File: Powerfull Integer[30-April-2023]/Solution.java
//  >> JAVA CODE <<

class Solution{
    public static int powerfullInteger(int n,int intervals[][],int k)
    {
        int mx = Integer.MIN_VALUE;
        for (int i = 0; i < n; i++) {
            mx = Math.max(mx, intervals[i][1]);
        }
        
        int[] v = new int[mx + 3];
        for (int i = 0; i < n; i++) {
            v[intervals[i][0]] += 1;
            v[intervals[i][1] + 1] -= 1;
        }
        
        int ans = -1;
        for (int i = 1; i < mx + 3; i++) {
            v[i] = v[i - 1] + v[i];
            if (v[i] >= k) {
                ans = i;
            }
        }
        return ans;
   

File: Powerfull Integer[30-April-2023]/solution.cpp
//   >> C++ CODE <<


int powerfullInteger(int n,vector<vector<int>> &intervals,int k){
        int mx=INT_MIN;
        for(int i=0;i<n;i++){
            mx=max(mx,intervals[i][1]);
        }
        vector<int>v(mx+3,0);
        for(int i=0;i<n;i++){
            v[intervals[i][0]]+=1;
            v[intervals[i][1]+1]-=1;
            
        }
        int ans=-1;
        for(int i=1;i<mx+3;i++){
             v[i]=v[i-1]+v[i];
             if(v[i]>=k) ans=i;
        }
        return ans;
    }